staging/src/k8s.io/code-generator/cmd/validation-gen/targets.go

```diff
@@ -205,47 +200,40 @@ func GetTargets(context *generator.Context, args *Args) []generator.Target {
 
 		// Find the right input pkg, which might not be this one.
 		inputPath := pkgToInput[input]
-		typesPkg = context.Universe[inputPath]
+		// typesPkg is where the types that need validation are defined.
+		// Sometimes it is different from pkg. For example, kubernetes core/v1
+		// types are defined in k8s.io/api/core/v1, while the pkg which holds
+		// defaulter code is at k/k/pkg/api/v1.
+		typesPkg := context.Universe[inputPath]
 
 		// Figure out which types we should be considering further.
 		var rootTypes []*types.Type
 		for _, t := range typesPkg.Types {
 			if shouldCreateObjectValidationFn(t) {
 				rootTypes = append(rootTypes, t)
+			} else {
+				klog.V(6).InfoS("skipping type", "type", t)
 			}
 		}
 		// Deterministic ordering helps in logs and debugging.
 		slices.SortFunc(rootTypes, func(a, b *types.Type) int {
 			return cmp.Compare(a.Name.String(), b.Name.String())
 		})
 
-		validationFunctionTypes := map[*types.Type]*callNode{}
+		// Deterministic ordering helps in logs and debugging.
+		slices.SortFunc(rootTypes, func(a, b *types.Type) int {
+			return cmp.Compare(a.Name.String(), b.Name.String())
+		})
```

Comment:
@jpbetz follow up: Remove accidental redundant sort.

Reply:
Oh, I must have missed that

Reply:
easy peasy, I'll drop in a fix for a few of these details shortly.